tests/momento/simple_cache_client/test_scalar_async.py

```diff
@@ -9,274 +12,268 @@
+
+    async def with_bad_key_throws_exception(
+        client_async: SimpleCacheClientAsync, cache_name: str, key_validator
+    ) -> None:
+        response = await key_validator(cache_name, 1)
+        assert response.error_code == MomentoErrorCode.INVALID_ARGUMENT_ERROR
+        assert response.inner_exception.message == "Unsupported type for key: <class 'int'>"
+
+
+def a_connection_validator():
+    async def throws_authentication_exception_for_bad_token(
+        bad_token_credential_provider: EnvMomentoTokenProvider,
+        configuration: Configuration,
+        cache_name: str,
+        default_ttl_seconds: timedelta,
+        connection_validator,
+    ) -> None:
+        async with SimpleCacheClientAsync(
+            configuration, bad_token_credential_provider, default_ttl_seconds
+        ) as client_async:
+            response = await connection_validator(client_async, cache_name)
+            assert response.error_code == MomentoErrorCode.AUTHENTICATION_ERROR
+
+    async def throws_timeout_error_for_short_request_timeout(
+        configuration: Configuration,
+        credential_provider: EnvMomentoTokenProvider,
+        cache_name: str,
+        default_ttl_seconds: timedelta,
+        connection_validator,
+    ) -> None:
+        configuration = configuration.with_client_timeout(timedelta(milliseconds=1))
+        async with SimpleCacheClientAsync(configuration, credential_provider, default_ttl_seconds) as client_async:
+            response = await connection_validator(client_async, cache_name)
+            assert response.error_code == MomentoErrorCode.TIMEOUT_ERROR
+
+
+def describe_set_and_get():
+    async def with_hit(client_async: SimpleCacheClientAsync, cache_name: str) -> None:
+        key = uuid_str()
+        value = uuid_str()
+
+        set_resp = await client_async.set(cache_name, key, value)
+        assert isinstance(set_resp, CacheSet.Success)
+
+        get_resp = await client_async.get(cache_name, key)
+        isinstance(get_resp, CacheGet.Hit)
+        assert get_resp.value_string == value
+        assert get_resp.value_bytes == str_to_bytes(value)
+
+    async def with_byte_key_values(client_async: SimpleCacheClientAsync, cache_name: str) -> None:
+        key = uuid_bytes()
+        value = uuid_bytes()
+
+        set_resp = await client_async.set(cache_name, key, value)
+        assert isinstance(set_resp, CacheSet.Success)
+
+        get_resp = await client_async.get(cache_name, key)
+        assert isinstance(get_resp, CacheGet.Hit)
+        assert get_resp.value_bytes == value
+
+
+@behaves_like(a_cache_name_validator)
+@behaves_like(a_key_validator)
+@behaves_like(a_connection_validator)
+def describe_get():
+    @fixture
+    def cache_name_validator(client_async: SimpleCacheClientAsync):
+        def _cache_name_validator(cache_name):
+            key = uuid_str()
+            return client_async.get(cache_name, key)
```

Comment:
At first I was unsure how we're invoking an async method (`client_async.get`) without the outer functions annotated as `async`. I suppose this is returning a coroutine object that is not on the event loop, and that is awaited in the caller functions above?